File: examples/ts/prediction/polymarket-fetch-event-by-tags.ts
import ccxt from '../../../js/ccxt.js';


// fetch polymarket events by tag — human-readable labels ("Fed Rates") or
// slugs ("fed-rates") both work; multiple tags match ANY of them

async function example () {
    const exchange = new ccxt.prediction.polymarket ();
    const events = await exchange.fetchEvents ({ 'tags': [ 'Fed Rates' ], 'limit': 5 });
    for (let i = 0; i < events.length; i++) {
        const event = events[i];
        console.log (event['title'], '| tags:', event['tags']);
        const markets = event['markets'];
        for (let j = 0; j < markets.length; j++) {
            const market = markets[j];
            const outcomes = market['outcomes'];
            for (let k = 0; k < outcomes.length; k++) {
                console.log ('   ', outcomes[k]['outcome'], '->', outcomes[k]['price']);
            }
        }
    }
    await exchange.close ();
}

await example ();

File: examples/py/prediction/polymarket-fetch-event-by-tags.py
# -*- coding: utf-8 -*-

# fetch polymarket events by tag — human-readable labels ("Fed Rates") or
# slugs ("fed-rates") both work; multiple tags match ANY of them

import asyncio
import os
import sys

root = os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))
sys.path.append(root + '/python')

import ccxt.prediction  # noqa: E402


async def main():
    exchange = ccxt.prediction.polymarket()
    try:
        events = await exchange.fetch_events({'tags': ['Fed Rates'], 'limit': 5})
        for event in events:
            print(event['title'], '| tags:', event['tags'])
            for market in event['markets']:
                for outcome in market['outcomes']:
                    print('   ', outcome['outcome'], '->', outcome['price'])
    finally:
        await exchange.close()


asyncio.run(main())
